app/src/main/java/com/omkarmoghe/pokemap/views/map/MapWrapperFragment.java

```diff
@@ -596,12 +596,12 @@ public void onAnimationStart(Animator animator) {
 
             @Override
             public void onAnimationEnd(Animator animator) {
+                pokemonMarker.getMarker().remove();
+                markerList.remove(pokemonMarker);
             }
 
             @Override
             public void onAnimationCancel(Animator animator) {
-                pokemonMarker.getMarker().remove();
```

Comment:
@s7092910 : Let do the test for it, and you will see how onAnimationCancel() be invoked. It's only called if you call animator.cancel(); when activity onPause or be force close, onAnimationCancel() callback won't be called.